### PR TITLE
hud module intersects with dev stats window

### PR DESCRIPTION
**Pull request by druiz17**
hud module intersects with dev stats window

### CODE REVIEW

**Comment by marko8904** [comment]
build this please

**Comment by druiz17** [comment]
Test plan
- In HMD mode open the stats window from the developer menu
- make sure that the laser interacts with the stats window.

**Comment by YuppyPlays** [comment]
Awaiting build for QA.

File: interface/resources/qml/Stats.qml [hunk 1]
```diff
@@ -8,6 +8,7 @@ Item {
 
     anchors.leftMargin: 300
     objectName: "StatsItem"
+    property var statsObject: root
```
Comment (SamGondelman) [inline interface/resources/qml/Stats.qml]: Is there a way to generalize this to all window-less qml windows?  I think things like the audio meter will still be broken.

File: interface/resources/qml/desktop/Desktop.qml [hunk 2]
```diff
@@ -301,14 +301,18 @@ FocusScope {
     function isPointOnWindow(point) {
         for (var i = 0; i < desktop.visibleChildren.length; i++) {
             var child = desktop.visibleChildren[i];
-            if (child.visible) {
```
Comment (SamGondelman) [inline interface/resources/qml/desktop/Desktop.qml]: I think we still want to do this visible check.

Comment (druiz17) [reply interface/resources/qml/desktop/Desktop.qml]: I removed that check.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4057/HighFidelity-Beta-PR11573-8f3fa08b1eae683812b1dbf3bf7097493af43b3d.dmg

File: interface/resources/qml/desktop/Desktop.qml (hunk 2 above)
Comment (SamGondelman) [reply interface/resources/qml/desktop/Desktop.qml]: Ah, nice.

File: interface/resources/qml/Stats.qml (hunk 1 above)
Comment (druiz17) [reply interface/resources/qml/Stats.qml]: I could try to convert stats and audio meter to be actual qml windows rather then just qml items

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4057/HighFidelity-Beta-PR11573-8f3fa08b1eae683812b1dbf3bf7097493af43b3d.exe

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
Testing complete:

In HMD mode open the stats window from the developer menu **Check**
make sure that the laser interacts with the stats window. **Check, Clicking trigger whilst aimed at stats window acts as a mouse click would in Desktop.**

**Notes: The collision for the laser goes around a square boundary for the whole stats window (which is comprised of smaller windows itself) which isn't an issue but, just thought I'd mention it just in-case.**

**Pass**

**Comment by druiz17** [comment]
@YuppyPlays I changed somethings can I get this QA again. I updated the test plan.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4069/HighFidelity-Beta-PR11573-42cab19ab2b14feaef66bbf41ebe1eaf1916deac.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4069/HighFidelity-Beta-PR11573-42cab19ab2b14feaef66bbf41ebe1eaf1916deac.exe

**Comment by YuppyPlays** [comment]
@druiz17 Absolutely no problem :)

Testing.

**Comment by YuppyPlays** [comment]
Testing complete:

All **Passes**  same as previous test :)

**Comment by SamGondelman** [comment]
It seems like there's a vertical offset on the stats and audio level meters between the rendering surface and the triggerable area.  And it would be nice if the non-visible area of the stats window didn't collide, like @YuppyPlays mentioned.

**Comment by druiz17** [comment]
@YuppyPlays I fixed some issues. Can I get this QA again. Follow the test plan above.

**Comment by YuppyPlays** [comment]
Testing.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4100/HighFidelity-Beta-PR11573-83d2a54b0fa132949b4eb3dd8b08a8fcac87e28c.dmg

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4104/HighFidelity-Beta-PR11573-83d2a54b0fa132949b4eb3dd8b08a8fcac87e28c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4104/HighFidelity-Beta-PR11573-83d2a54b0fa132949b4eb3dd8b08a8fcac87e28c.exe

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
Testing complete:

In HMD mode open the stats window from the developer menu **Check**
also bring up the audio meter by enabling 'show audio level meter' from the audio app **Check**
make sure that the laser interacts with the stats window and the audio meter. **Check**
while the stats window is not visible the laser should not interact with the area where the stats window was at. **Check**

**Notes: All in order, though the same reported possible issue is how the laser works around the whole stats box, as opposed to just hitting each individual one, so this obscures laser usability within that region of the players view, though I do not think this is a massive issue, just feel it is worth mentioning.**

**Pass**

File: interface/resources/qml/desktop/Desktop.qml [hunk 3]
```diff
@@ -301,15 +301,20 @@ FocusScope {
...
-                if (child.hasOwnProperty("modality")) {
-                    var mappedPoint = child.mapFromGlobal(point.x, point.y);
+            if (child.hasOwnProperty("modality")) {
+                var mappedPoint = mapToItem(child, point.x, point.y);
+                if (child.hasOwnProperty("frame")) {
                     var outLine = child.frame.children[2];
                     var framePoint = outLine.mapFromGlobal(point.x, point.y);
                     if (child.contains(mappedPoint) || outLine.contains(framePoint)) {
+                        console.log("it should not be here");
```
Comment (SamGondelman) [inline interface/resources/qml/desktop/Desktop.qml]: Leftover logging, and you could get rid of the `child.contains(mappedPoint)` here and just have it fall through to the check below.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4126/HighFidelity-Beta-PR11573-2422c9d9f3efb5bd88ecbc7c391a84b506ef017f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4126/HighFidelity-Beta-PR11573-2422c9d9f3efb5bd88ecbc7c391a84b506ef017f.exe

**Pull request merged**